### PR TITLE
Merge RC functionality part1

### PR DESCRIPTION
**Pull request by AByzhynar**
This PR provides the following changes:

Main CMakeLists.txt and MD files' changes
- Added necessary flags, variables adn their handling to make possible to
  build SDL with RC functionality
- Updated COMMITERS.md with actual information

Config Profile's changes:  
- Updated config profle to support plugin folder

Connection hamdler's changes:
- Updated connection handler OnDeviceRemove() method

Formatters' & Smart Objects' changes: 
 - Removed redundant and commented code from formatters & smart objects unit tests 

Security Manager's changes: 
 - Added mandatory header files includes to SSL context
 - Added funcion GetName to get certificate name and some logging to SSL
    context
 - Added function SetCertExpTime and updated appropriate mock classes
    
Media manager's changes: 
 - Updated header guards
 - Included necessary paths, added libriries and added
    RC fuctionality support to CMakeLists
 - Added REMOTE_CONTROL handling to CmakeLists
 - Replaced raw ptrs with SharedPtr in setters for unit tests
 - Added/updated unit tests for media manager

Utils' changes: 
  - Added explicit header <fstream>
  - Removed unused preprocessor directives handling of __QNXNTO__
  - Added export function definition
  - Added case sensitive comparison methods
  - Update UT macro FRIEND_TEST
  - Added timer_thread class
  - Fixed memory leaks
  - Updated utils unit tests

Telemetry Monitor's changes:
  - Added TimeTester library creation and included RC functionality
    directories
  - Fixed memory leaks
 
Protocol handler's changes:
  - Added service type check and named parameter protocol_version
  
Transpport Manager
  - Enabled previously disabled nit test
  
Updated auxiliary tools

### CODE REVIEW

**Comment by AByzhynar** [comment]
@LuxoftAKutsan @anosach-luxoft @dev-gh @AKalinich-Luxoft Please review

File: CMakeLists.txt [hunk 1]
```diff
@@ -109,8 +110,11 @@ if (MEDIA_MODE_OPTION)
   endif()
 endif()
 
-if( NOT CMAKE_BUILD_TYPE )
-  set(CMAKE_BUILD_TYPE Debug CACHE STRING "Choose the type of build. Options are: None, Debug, Release, RelWithDebInfo, MinSizeRel." FORCE)
+if (DEBUG_OPTION)
```
Comment (LuxoftAKutsan) [inline CMakeLists.txt]: @AByzhynar this changes are redundant

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -25,7 +25,7 @@ We use <a href="http://nvie.com/posts/a-successful-git-branching-model/">Gitflow
 * A great example of a [pull request can be found here](https://github.com/smartdevicelink/SmartDeviceLink-iOS/pull/45).
 
 ### Contributor's License Agreement (CLA)
-In order to accept Pull Requests from contributors, you must first sign [the Contributor's License Agreement](https://docs.google.com/forms/d/1VNR8EUd5b46cQ7uNbCq1fJmnu0askNpUp5dudLKRGpU/viewform). If you need to make a change to information that you entered, [please contact us](mailto:theresa@livio.io).
+In order to accept Pull Requests from contributors, you must first sign [the Contributor's License Agreement](https://docs.google.com/forms/d/1VNR8EUd5b46cQ7uNbCq1fJmnu0askNpUp5dudLKRGpU/viewform). If you need to make a change to information that you entered, [please contact us](mailto:justin@livio.io).
```
Comment (LuxoftAKutsan) [inline CONTRIBUTING.md]: @AByzhynar please remove changes.

File: src/components/config_profile/include/config_profile/profile.h [hunk 3]
```diff
@@ -154,10 +154,12 @@ class Profile : public protocol_handler::ProtocolHandlerSettings,
    */
   const uint16_t video_streaming_port() const OVERRIDE;
 
+  const uint16_t audio_streaming_port() const OVERRIDE;
```
Comment (LuxoftAKutsan) [inline src/components/config_profile/include/config_profile/profile.h]: @AByzhynar description missed

File: src/components/config_profile/include/config_profile/profile.h [hunk 4]
```diff
@@ -344,10 +346,19 @@ class Profile : public protocol_handler::ProtocolHandlerSettings,
...
 
+  /*
+   * @brief Returns true if last state singleton is used
+   */
   bool use_last_state() const OVERRIDE;
 
+  /**
+    * @brief Returns port for audio streaming
+    */
   uint32_t transport_manager_disconnect_timeout() const OVERRIDE;
```
Comment (LuxoftAKutsan) [inline src/components/config_profile/include/config_profile/profile.h]: @AByzhynar wrong description

File: src/components/config_profile/src/profile.cc [hunk 5]
```diff
@@ -965,6 +970,8 @@ void Profile::UpdateValues() {
 
   LOG_UPDATED_BOOL_VALUE(logs_enabled_, kLogsEnabledKey, kMainSection);
 
+  logs_enabled_ = true;
```
Comment (LuxoftAKutsan) [inline src/components/config_profile/src/profile.cc]: @AByzhynar please remove this line.

File: src/components/config_profile/test/profile_test.cc [hunk 6]
```diff
@@ -744,6 +744,6 @@ TEST_F(ProfileTest, AppIconsFolder_DefaultValues_SUCCESS) {
             profile_.app_icons_folder_max_size());
 }
 
-}  // namespace profile
+}  // namespace profile_test
```
Comment (LuxoftAKutsan) [inline src/components/config_profile/test/profile_test.cc:747]: @AByzhynar please check check if actual name of namespace is profile_test

File: src/components/include/security_manager/crypto_manager.h [hunk 7]
```diff
@@ -71,6 +71,7 @@ class CryptoManager : public policy::PolicyHandlerObserver {
   * \return pointer to crypto manager settings class
   */
   virtual const CryptoManagerSettings& get_settings() const = 0;
+  virtual void SetCertExpTime(const std::string& time) = 0;
```
Comment (LuxoftAKutsan) [inline src/components/include/security_manager/crypto_manager.h]: @AByzhynar  please check if this method used

File: src/components/include/utils/timer_thread.h [hunk 8]
```diff
@@ -0,0 +1,421 @@
+/*
```
Comment (LuxoftAKutsan) [inline src/components/include/utils/timer_thread.h:1]: @AByzhynar please check if htis file used

File: src/components/media_manager/CMakeLists.txt [hunk 9]
```diff
@@ -31,18 +31,76 @@
 include(${CMAKE_SOURCE_DIR}/tools/cmake/helpers/platform.cmake)
 include(${CMAKE_SOURCE_DIR}/tools/cmake/helpers/sources.cmake)
 
-include_directories(
-    include
+if (EXTENDED_MEDIA_MODE)
+find_package(Gstreamer-1.0 REQUIRED)
```
Comment (LuxoftAKutsan) [inline src/components/media_manager/CMakeLists.txt]: @AByzhynar I am nut sure that is required changes. Please remove it.

File: src/components/media_manager/CMakeLists.txt [hunk 10]
```diff
@@ -31,18 +31,76 @@
...
+set(LIBRARIES
+    ProtocolLibrary
+)
+endif()
+
+include_directories (
+    ${COMPONENTS_DIR}/media_manager/include/
     ${COMPONENTS_DIR}/media_manager/include/audio/
     ${COMPONENTS_DIR}/media_manager/include/video/
+    ${COMPONENTS_DIR}/can_cooperation/include/
```
Comment (LuxoftAKutsan) [inline src/components/media_manager/CMakeLists.txt]: @AByzhynar this changes required

File: src/components/media_manager/test/media_manager_impl_test.cc [hunk 11]
```diff
@@ -33,147 +33,376 @@
...
+       StartMicrophoneRecording_SourceFileIsWritable_ExpectTrue) {
+  StartMicrophoneCheckHelper();
+  if (!DirectoryExists(kStorageFolder)) {
+    CreateDirectory(kStorageFolder);
+  }
+  EXPECT_FALSE(FileExists(kOutputFilePath));
+  EXPECT_TRUE(CreateFile(kOutputFilePath));
 
-  mediaManagerImpl.set_mock_mic_listener(media_adapter_listener_mock_);
+  const std::string record_file_path =
```
Comment (LuxoftAKutsan) [inline src/components/media_manager/test/media_manager_impl_test.cc:298]: @AByzhynar please remove changes related to extended media mode

File: src/components/security_manager/src/ssl_context_impl.cc [hunk 12]
```diff
@@ -136,23 +138,33 @@ std::map<std::string, CryptoManagerImpl::SSLContextImpl::BlockSizeGetter>
...
-      OPENSSL_free(subj);
-    }
-    char* issuer = X509_NAME_oneline(X509_get_issuer_name(cert), NULL, 0);
-    if (issuer) {
-      std::replace(issuer, issuer + strlen(issuer), '/', ' ');
-      LOG4CXX_DEBUG(logger_, cert_owner << " issuer:" << issuer);
-      OPENSSL_free(issuer);
-    }
-
+    LOG4CXX_DEBUG(logger_,
```
Comment (LuxoftAKutsan) [inline src/components/security_manager/src/ssl_context_impl.cc:162]: @AByzhynar are you sure, that this code should be deleted

**Review by LuxoftAKutsan — CHANGES_REQUESTED** [review]

File: src/components/config_profile/test/profile_test.cc (hunk 6 above)
Comment (AByzhynar) [reply src/components/config_profile/test/profile_test.cc:747]: @LuxoftAKutsan It is actual. Checked.

File: src/components/include/security_manager/crypto_manager.h (hunk 7 above)
Comment (AByzhynar) [reply src/components/include/security_manager/crypto_manager.h]: @LuxoftAKutsan It was added by this PR: https://github.com/smartdevicelink/sdl_core/pull/1434
Do you suggest - I have to remove all changes provided by this PR?

File: src/components/security_manager/src/ssl_context_impl.cc (hunk 12 above)
Comment (AByzhynar) [reply src/components/security_manager/src/ssl_context_impl.cc:162]: @LuxoftAKutsan If I understand correctly this code was not removed but just extracted to separate function and used here.

**Review by AKalinich-Luxoft — APPROVED** [review]

File: src/components/CMakeLists.txt [hunk 13]
```diff
@@ -86,9 +86,16 @@ add_subdirectory(./config_profile)
...
+  add_subdirectory(./can_cooperation)
+  # --- Functional module
+  add_subdirectory(./functional_module)
+endif()
+
 # --- Telemetry Monitor
-if(TELEMETRY_MONITOR)
-  add_subdirectory(./telemetry_monitor)
+if (TELEMETRY_MONITOR)
+    add_subdirectory(./telemetry_monitor)
```
Comment (dev-gh) [inline src/components/CMakeLists.txt]: @AByzhynar 
I think formatting should be fixed

File: CMakeLists.txt [hunk 14]
```diff
@@ -240,10 +248,10 @@ if(ENABLE_SANITIZE)
   message(STATUS "Sanitizing enabled")
 endif()
 
-if(ENABLE_GCOV)
+if (ENABLE_GCOV)
```
Comment (dev-gh) [inline CMakeLists.txt]: @AByzhynar 
Not related to RC

File: src/components/application_manager/test/commands/mobile/show_test.cc [hunk 15]
```diff
@@ -765,6 +765,10 @@ TEST_F(ShowRequestTest, OnEvent_SuccessResultCode_SUCCESS) {
 
   SharedPtr<ShowRequest> command(CreateCommand<ShowRequest>(msg));
 
+  EXPECT_CALL(mock_message_helper_,
```
Comment (dev-gh) [inline src/components/application_manager/test/commands/mobile/show_test.cc:768]: @AByzhynar 
Don't think its related to RC

File: src/components/connection_handler/src/connection_handler_impl.cc [hunk 16]
```diff
@@ -169,12 +169,12 @@ void ConnectionHandlerImpl::OnDeviceRemoved(
...
   }
 
-  device_list_.erase(device_info.device_handle());
   sync_primitives::AutoReadLock read_lock(connection_handler_observer_lock_);
   if (connection_handler_observer_) {
     connection_handler_observer_->RemoveDevice(device_info.device_handle());
     connection_handler_observer_->OnDeviceListUpdated(device_list_);
   }
+  device_list_.erase(device_info.device_handle());
 }
```
Comment (dev-gh) [inline src/components/connection_handler/src/connection_handler_impl.cc:178]: @AByzhynar 
Don't think its related to RC

File: src/components/include/test/media_manager/mock_media_manager.h [hunk 17]
```diff
@@ -29,8 +29,8 @@
  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  * POSSIBILITY OF SUCH DAMAGE.
  */
-#ifndef SRC_COMPONENTS_INCLUDE__TEST_MEDIA_MANAGER_MOCK_MEDIA_MANAGER_H_
-#define SRC_COMPONENTS_INCLUDE__TEST_MEDIA_MANAGER_MOCK_MEDIA_MANAGER_H_
+#ifndef SRC_COMPONENTS_INCLUDE_TEST_MEDIA_MANAGER_MOCK_MEDIA_MANAGER_H_
```
Comment (dev-gh) [inline src/components/include/test/media_manager/mock_media_manager.h:32]: @AByzhynar 
Don't think its related to RC as well as other media related changes

File: src/components/include/utils/custom_string.h [hunk 18]
```diff
@@ -180,6 +180,20 @@ class CustomString {
   bool CompareIgnoreCase(const char* str) const;
 
   /**
+   * @brief Compares the value of the string (case sensitive).
```
Comment (dev-gh) [inline src/components/include/utils/custom_string.h:183]: @AByzhynar 
Don't think its related to RC

File: src/components/security_manager/src/ssl_context_impl.cc [hunk 19]
```diff
@@ -136,23 +138,33 @@ std::map<std::string, CryptoManagerImpl::SSLContextImpl::BlockSizeGetter>
     CryptoManagerImpl::SSLContextImpl::max_block_sizes =
         CryptoManagerImpl::SSLContextImpl::create_max_block_sizes();
 
+std::string GetName(X509_NAME* x509_name) {
```
Comment (dev-gh) [inline src/components/security_manager/src/ssl_context_impl.cc:141]: @AByzhynar 
Don't think its related to RC

File: src/components/telemetry_monitor/CMakeLists.txt [hunk 20]
```diff
@@ -69,6 +71,14 @@ add_library("TelemetryMonitor" ${SOURCES})
 target_link_libraries("TelemetryMonitor" ${LIBRARIES})
 add_dependencies("TelemetryMonitor" HMI_API MOBILE_API)
 
+set (LIBRARIES
```
Comment (dev-gh) [inline src/components/telemetry_monitor/CMakeLists.txt:74]: @AByzhynar 
Don't think its related to RC

File: src/components/utils/src/file_system.cc [hunk 21]
```diff
@@ -100,9 +94,6 @@ size_t file_system::DirectorySize(const std::string& path) {
     }
   }
   closedir(directory);
-#ifdef __QNXNTO__
```
Comment (dev-gh) [inline src/components/utils/src/file_system.cc:103]: @AByzhynar 
Don't think its related to RC

File: src/components/formatters/test/meta_formatter_test.cc [hunk 22]
```diff
@@ -68,20 +68,6 @@ TEST_F(CMetaFormatterTestHelper,
       CMetaFormatter::CreateObjectByPattern(object2, schema, result_object2);
   EXPECT_TRUE(creationresult);
 
-  // Uncomment code to print objects in console
```
Comment (dev-gh) [inline src/components/formatters/test/meta_formatter_test.cc:71]: @AByzhynar 
I think task was to take only RC changes but this does not have to do with these changes

File: src/components/utils/src/file_system.cc (hunk 21 above)
Comment (AByzhynar) [reply src/components/utils/src/file_system.cc:103]: @dev-gh Right but these changes are critical and needed for further development.

File: src/components/telemetry_monitor/CMakeLists.txt (hunk 20 above)
Comment (AByzhynar) [reply src/components/telemetry_monitor/CMakeLists.txt:74]: @dev-gh Right but these changes are critical and needed for further development.

File: src/components/security_manager/src/ssl_context_impl.cc (hunk 19 above)
Comment (AByzhynar) [reply src/components/security_manager/src/ssl_context_impl.cc:141]: @dev-gh Right but these changes are critical and needed for further development.

File: src/components/include/utils/custom_string.h (hunk 18 above)
Comment (AByzhynar) [reply src/components/include/utils/custom_string.h:183]: @dev-gh Right but these changes are critical and needed for further development.

File: src/components/include/test/media_manager/mock_media_manager.h (hunk 17 above)
Comment (AByzhynar) [reply src/components/include/test/media_manager/mock_media_manager.h:32]: @dev-gh Right but these changes are critical and needed for further development.

File: src/components/formatters/test/meta_formatter_test.cc (hunk 22 above)
Comment (AByzhynar) [reply src/components/formatters/test/meta_formatter_test.cc:71]: @dev-gh No. The task is to combine RC changes with develop and with CRQs. And also these specific changes were discussed with @LuxoftAKutsan

File: src/components/connection_handler/src/connection_handler_impl.cc (hunk 16 above)
Comment (AByzhynar) [reply src/components/connection_handler/src/connection_handler_impl.cc:178]: @dev-gh Right but these changes are critical and needed for further development.

File: src/components/application_manager/test/commands/mobile/show_test.cc (hunk 15 above)
Comment (AByzhynar) [reply src/components/application_manager/test/commands/mobile/show_test.cc:768]: @dev-gh Right but these changes are critical and needed for further development.

File: src/components/CMakeLists.txt (hunk 13 above)
Comment (AByzhynar) [reply src/components/CMakeLists.txt]: @dev-gh Will be changed back

**Review by dev-gh — APPROVED** [review]

**Comment by AByzhynar** [comment]
@LuxoftAKutsan All changes done. Commits amended. Please check.

**Review by LuxoftAKutsan — APPROVED** [review]

**Pull request merged**